Makefile

```diff
@@ -72,71 +71,57 @@ essr: VERSION
-	@cd $(ESSDIR)/doc/ ; $(MAKE) html
-	@echo "** Creating .tgz file **"
+	@echo "** Making distribution of ESS for (pre)release $(ESSVERSION) from $(ESSDIR)/"
 	test -f $(ESSDIR).tgz && rm -rf $(ESSDIR).tgz || true
+	@echo "** Creating .tgz file **"
 	$(GNUTAR) hcvofz $(ESSDIR).tgz $(ESSDIR)
 	@echo "Signing tgz file"
-	$(GPG) -ba -o $(ESSDIR).tgz.sig $(ESSDIR).tgz
+	-$(GPG) -ba -o $(ESSDIR).tgz.sig $(ESSDIR).tgz
 	@echo "** Creating .zip file **"
 	test -f $(ESSDIR).zip && rm -rf $(ESSDIR).zip || true
 	zip -r $(ESSDIR).zip $(ESSDIR)
 	@echo "Signing zip file"
-	$(GPG) -ba -o $(ESSDIR).zip.sig $(ESSDIR).zip
-	touch $@
+	-$(GPG) -ba -o $(ESSDIR).zip.sig $(ESSDIR).zip
 
 # Create the "release" directory
 # run in the foreground so you can accept the certificate
-# NB 'all', 'cleanup-dist' must not be targets: otherwise, e.g.
-#    'make tarball' re-builds the tarballs always!
-$(ESSDIR): RPM.spec
-	$(MAKE) all
-#	remove previous ESSDIR, etc:
-	$(MAKE) cleanup-dist
+# NB 'all', 'cleanup-rel' must not be targets: otherwise, e.g.
+#    'make tarball' re-builds the tarballs always!  (??? -- dickmao)
+.PHONY: rel-staging
+rel-staging: cleanup-rel
 	@echo "**********************************************************"
 	@echo "** Making $(ESSDIR) directory of ESS for release $(ESSVERSION),"
 	@echo "** (must have setup git / github with cached authentication, prior for security)"
 	@echo "**********************************************************"
 	@echo "** Exporting Files **"
-	git clone git@github.com:emacs-ess/ESS.git $(ESSDIR)-git
+	mkdir -p $(ESSDIR)-git
+	stash=`git stash create`; git archive --format tar $${stash:-HEAD} | (cd $(ESSDIR)-git ; $(GNUTAR) xf -)
 	mkdir -p $(ESSDIR)
 	(cd $(ESSDIR)-git; $(GNUTAR) cvf - --exclude=.git --exclude=.svn --no-wildcards .) | (cd $(ESSDIR); $(GNUTAR) xf - )
-	@echo "** Clean-up docs, Make docs, and Correct Write Permissions **"
-	CLEANUP="user-* useR-* Why_* README.*"; ED=$(ESSDIR)/doc; \
-	 if [ -d $$ED ] ; then CD=`pwd`; cd $$ED; chmod -R u+w $$CLEANUP; rm -rf $$CLEANUP; \
-	 $(MAKE) all cleanaux ; cd $$CD; fi
-#   just in case: update from VERSION:
-	cd lisp; $(INSTALL) ess.el ../$(ESSDIR)/lisp/; cd ..
-	cd lisp; $(MAKE) julia-mode.el; $(INSTALL) julia-mode.el ../$(ESSDIR)/lisp/; cd ..
-	$(INSTALL) RPM.spec $(ESSDIR)/
+	cd $(ESSDIR) ; $(MAKE) dist
+	cd $(ESSDIR)/doc ; $(MAKE) cleanaux
+	cd lisp; $(MAKE) ess-custom.el; $(INSTALL) ess-custom.el ../$(ESSDIR)/lisp/
+	cd lisp; $(MAKE) julia-mode.el; $(INSTALL) julia-mode.el ../$(ESSDIR)/lisp/
 	chmod a-w $(ESSDIR)/lisp/*.el
 	chmod u+w $(ESSDIR)/lisp/ess-site.el $(ESSDIR)/Make* $(ESSDIR)/*/Makefile
 	touch $(ESSDIR)/etc/.IS.RELEASE
 #	# Get (the first 12 hexdigits of) the git version into the release tarball:
-	cut -c 1-12 $(ESSDIR)-git/.git/refs/heads/master > $(ESSDIR)/etc/git-ref
-
-dist: VERSION tarballs
-	grep -E 'defvar ess-(version|revision)' lisp/ess-custom.el \
-	  $(ESSDIR)/lisp/ess-custom.el > $@
+	if git diff-index --quiet HEAD -- ; then \
```

Comment:
Could you please describe for mortals what this one is doing and why it is needed?

Reply:
You git clone master.  You change VERSION.  You run `make tarballs` to see if it takes.

It doesn't because [you're still getting the Makefile from master](https://github.com/emacs-ess/ESS/blob/47bd3d257f99eec427d296226826e1c1b45919bd/Makefile#L107).

The line in question says: "If you have made a change in your working tree, create a new git-ref and pack it into the tarball.  If you have not made any changes, pack the master's git-ref."